### ISSUE TITLE
Fix `media-feature-name-no-unknown` false positives for `not` keyword

### ISSUE THREAD

**Issue by gijsk** [stylelint/stylelint#6834]
### What steps are needed to reproduce the bug?

```css
@media screen and (not (prefers-contrast)) {
  p { font: bold }
}
```

### What Stylelint configuration is needed to reproduce the bug?

```js
module.exports = {
  extends: ["stylelint-config-recommended"],
};
```

### How did you run Stylelint?

https://stylelint.io/demo/#N4Igxg9gJgpiBcIACBbGUCWBDABAZzACcYYA7HLUqHAClIgBdaAHYgMxkLwFpJSHCWPAwCUInMAA65HMwk42EfvBwAjCABtqAX2m7SIADQg2GDTAByWNAhAwAHtebmAdGDx4j4JaYDmtqRlJO3sGMig8YJUAbWkceJxg4QBPcw0Mfl4fDF9uYUooLEIoYLicAF09Lz4-ADEIQhQsBlsAKzwlL1hmT0RA+KSGVJh0-ijEkA1mmGFgwzLB4dGGLNI-PIYCopKECamw2ZAq7SA

### Which version of Stylelint or dependencies are you using?

Locally,

"stylelint": "^15.6.0",
"stylelint-config-recommended": "^12.0.0",

In the demo, whatever the demo is using.

### What did you expect to happen?

No errors.

### What actually happened?

ERROR
Unexpected unknown media feature name "not (prefers-contrast)"
[(media-feature-name-no-unknown)](https://stylelint.io/user-guide/rules/media-feature-name-no-unknown)
[1:20-42]

### Does the bug relate to non-standard syntax?

Not non-standard syntax, I believe. This parses fine (tested using `window.matchMedia` in a devtools console) in both Firefox and Chrome. I think it's reliant on Level 4 Media Query support.

**Comment by ybiquitous** [comment]
@gijsk Thanks for the report with a reproducible demo! I can reproduce the bug. 👍🏼 

> I think it's reliant on Level 4 Media Query support.

I also agree that the `media-feature-name-no-unknown` rule should consider the `not` keyword (maybe also `only`) defined in [Media Queries Level 4](https://www.w3.org/TR/mediaqueries-4/#mq-syntax).

I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by romainmenke** [comment]
I think this is a good candidate for our media query parser.
I will see if I can migrate this rule.
